request-cert/main.go

```diff
@@ -52,53 +51,103 @@ func main() {
 
 	// Check certificate type.
 	var template *x509.CertificateRequest
-	var filePrefix string
+	var filename, csrName string
 	var wantServerAuth bool
 
+	hostname, err := os.Hostname()
```

Comment:
you're definitely correct. We do want to fail duplicate CSRs (eg: two client nodes initializing at the same time) but we should let things run in separate namespaces properly.
I still need to test out secret readability when running in different namespaces. My understanding is that you're supposed to be able to access them.
I'll do this before merging this.